src/LSP/capabilities/server.jl

```diff
@@ -0,0 +1,100 @@
+"""
+Since version 3.6.0
+
+Many tools support more than one root folder per workspace. Examples for this are VS Code’s
+multi-root support, Atom’s project folder support or Sublime’s project support. If a client
+workspace consists of multiple roots then a server typically needs to know about this.
+The protocol up to now assumes one root folder which is announced to the server by the
+`rootUri` property of the `InitializeParams`. If the client supports workspace folders and
+announces them via the corresponding `workspaceFolders` client capability,
+the `InitializeParams` contain an additional property `workspaceFolders` with the configured
+workspace folders when the server starts.
+
+The `workspace/workspaceFolders` request is sent from the server to the client to fetch the
+current open list of workspace folders.
+Returns null in the response if only a single file is open in the tool.
+Returns an empty array if a workspace is open but no folders are configured.
+"""
+@interface WorkspaceFoldersServerCapabilities begin
+    "The server has support for workspace folders"
+    supported::Union{Bool, Nothing} = nothing
+
+    """
+    Whether the server wants to receive workspace folder change notifications.
+
+    If a string is provided, the string is treated as an ID under which the notification is
+    registered on the client side. The ID can be used to unregister for these events using
+    the `client/unregisterCapability` request.
+    """
+    changeNotifications::Union{Union{String, Bool}, Nothing} = nothing
+end
+
+@interface ServerCapabilities begin
+    """
+    The position encoding the server picked from the encodings offered by the client via
+    the client capability `general.positionEncodings`.
+
+    If the client didn't provide any position encodings the only valid value that a server
+    can return is 'utf-16'. If omitted it defaults to 'utf-16'.
+
+    # Tags
+    - since – 3.17.0
+    """
+    positionEncoding::Union{PositionEncodingKind.Ty, Nothing} = nothing
+
+    """
+    Defines how text documents are synced. Is either a detailed structure defining each
+    notification or for backwards compatibility the TextDocumentSyncKind number.
+    If omitted it defaults to `TextDocumentSyncKind.None`.
+    """
+    textDocumentSync::Union{Union{TextDocumentSyncOptions,
+                                  TextDocumentSyncKind.Ty},
+                            Nothing} = nothing
```

Comment:
I'm not sure if I like this syntax. Maybe just make this one line?

Reply:
I wanted to adhere to the 92 character limit mentioned in the [formatting guidelines](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#general-formatting-guidelines-for-julia-code-contributions). Wasn't sure how to do that in a nice way though. I can keep it in one line if you prefer. Or maybe you have another preference that splits the line still?